### ISSUE TITLE
Chezmoi doesn't use Cygwin home path

### ISSUE THREAD

**Issue by lmf-mx** [twpayne/chezmoi#977]
## Describe the bug

Chezmoi doesn't use Cygwin's home when called from Cygwin, but instead uses the Windows user's user folder.

## To reproduce

Install
```
$ chezmoi init
Initialized empty Git repository in /cygdrive/c/Users/<user>/.local/share/chezmoi/.git/
```

## Expected behavior

Initialized empty Git repository in /home/<user>/.local/share/chezmoi/.git/

## Output of command with `--verbose`

```
mkdir -m 777 'C:\\Users\\<user>\\.local\\share': mkdir C:\Users\<user>\.local\share: The system cannot find the path specified.
mkdir -m 777 'C:\\Users\\<user>\\.local'
mkdir -m 777 'C:\\Users\\<user>\\.local\\share'
mkdir -m 700 'C:\\Users\\<user>\\.local\\share\\chezmoi'
Initialized empty Git repository in /cygdrive/c/Users/<user>/.local/share/chezmoi/.git/
( cd 'C:\\Users\\<user>\\.local\\share\\chezmoi' && 'C:\\cygwin64\\bin\\git.exe' init )e
```

## Output of `chezmoi doctor`

```
     ok: version 1.8.10, commit 11ed57d9a7e86434744f336b595ed0fd19aff6f5, built at 2020-12-20T12:09:31Z, built by goreleaser
     ok: runtime.GOOS windows, runtime.GOARCH amd64
  ERROR: C:\Users\<user>\.local\share\chezmoi (source directory, perm 777)
     ok: C:\Users\<user> (destination directory, perm 777)
warning: C:\Users\<user>\.config\chezmoi\chezmoi.toml (configuration file)
  ERROR: /bin/bash (shell, not found)
     ok: C:\cygwin64\bin\vi.exe (editor)
warning: vimdiff (merge command, not found)
     ok: C:\cygwin64\bin\git.exe (source VCS command, version 2.29.2)
warning: C:\cygwin64\bin\gpg.exe (GnuPG)
warning: op (1Password CLI, not found)
warning: bw (Bitwarden CLI, not found)
warning: gopass (gopass CLI, not found)
warning: keepassxc-cli (KeePassXC CLI, not found)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

## Additional context

64-bit Cygwin on Win10 64-bit

**Comment by twpayne** [comment]
Thank you for reporting this.

chezmoi uses the Go standard library's [`os.UserHomeDir`](https://pkg.go.dev/os#UserHomeDir) function to determine your home directory. This uses `%USERPROFILE%` on Windows, which is presumably set in Cygwin (it's been a long time since I used Cygwin, so please correct any errors that I make).

You can override chezmoi's destination directory with the `-D` flag. Specifically, I suggest running:

    chezmoi init -D $HOME

This should prioritize Cygwin's `$HOME` environment variable over Windows' `%USERPROFILE%`.

Please let me know if this works.

**Comment by lmf-mx** [comment]
I think that would be an acceptable workaround that should be documented. Unfortunately, however, the destination option doesn't work at all.

Even in Windows Powershell, I get this
```
PS C:\Users\<user>> chezmoi init -D 'C:\temp' --verbose
mkdir -m 777 'C:\\Users\\<user>\\.local\\share': mkdir C:\Users\<user>\.local\share: The system cannot find the path specified.
mkdir -m 777 'C:\\Users\\<user>\\.local'
mkdir -m 777 'C:\\Users\\<user>\\.local\\share'
mkdir -m 700 'C:\\Users\\<user>\\.local\\share\\chezmoi'
Initialized empty Git repository in C:/Users/<user>/.local/share/chezmoi/.git/
( cd 'C:\\Users\\<user>\\.local\\share\\chezmoi' && 'C:\\Program Files\\Git\\cmd\\git.exe' init )
```

Also, I tried --debug and no file is created, only standard verbose output to screen.

**Comment by twpayne** [comment]
Thank you for the further investigation. This sounds like a genuine bug.

**Comment by lmf-mx** [comment]
So, I should have been using the source switch, it seems.

Notice the peculiar output.
```
$ chezmoi init -S $(cygpath -w /home/<user>/.privatedot) --verbose
mkdir -m 777 'C:\\cygwin64\\home\\<user>': mkdir C:\cygwin64\home\lfudala: Cannot create a file when that file already exists.
mkdir -m 700 'C:\\cygwin64\\home\\<user>\\.privatedot'
Initialized empty Git repository in /home/<user>/.privatedot/.git/
( cd 'C:\\cygwin64\\home\\<user>\\.privatedot' && 'C:\\cygwin64\\bin\\git.exe' init )
```

I have to give chezmoi the path in a Windows format, but since it finds the Cygwin distributed git first on the path, it returns the directory in Cygwin's format.

What I did was take a step back and created the config file upfront with sourceDir and destDir. Then, alias chezmoi to always use that config.

I ended up creating `/home/<user>/.config/chezmoi/chezmoi.yaml` and using the alias
```
alias chezmoi="chezmoi -c '$(cygpath -w /home/$(whoami)/.config/chezmoi/chezmoi.yaml)'"
```

Then I was able to `chezmoi init` and `chezmoi add --encrypt <file>` and everything plays nicely together. I think a note in the docs about Cygwin pointing to this issue or detailing the path issues might help someone else who stumbles into this problem.

Note: To use gpg in Cygwin, I had to specify the path in the chezmoi config file to point to the gpg4win installation, since it doesn't pick up the Cygwin path, where I have `/usr/local/bin/gpg` as a symlink to it.
